File: Tests/Tests/RegisterPageTests.cs
using Atata;
using NUnit.Framework;

namespace Tests.Tests
{
    public class RegisterPageTests : BaseTest
    {
        [TestCase("Test", "password", "password", "some-email@test.com", "User with this login is already registered.")]
        [TestCase("Name", "password", "password", "test@test.com", "User with this email is already registered.")]
        [TestCase("Name", "password", "passwort", "some-email@test.com", "Passwords are different.")]
        [TestCase("Name", "password", "password", "@test.com", "Invalid email.")]
        [TestCase("Name", "password", "password", "some.test.com", "Invalid email.")]
        [TestCase("Name", "pass", "pass", "some@test.com", "Password is too short.")]
        public void RegisterWithInvalidData(string login, string password1, string password2, string email, string error) =>
            CreateDefaultUser()
                .OpenRegistration()
                .Register(login, email, password1, password2)
                .Error.Should.Equal(error);
    }
}

File: Tests/src/test/java/tests/RegisterPageTests.java
package test.java.tests;

import java.util.stream.Stream;

import org.junit.Assert;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

public class RegisterPageTests extends BaseTest {
	@Test
	public void positiveRegistrationTest() {
		String login = "LoGin";
		String password = "Password";
		String email = "login@test.com";
		
		String pageTitle = loginPage
			.openRegistration()
			.register(login.toLowerCase(), email, password)
			.login(login, password)
			.getTitle();
		
		Assert.assertEquals("Deposit calculator", pageTitle);
	}
	
	static Stream<Arguments> credentials(){
	    return Stream.of(
 	       Arguments.of("Test", "password", "password", "some-email@test.com", "User with this login is already registered."),
 	       Arguments.of("User", "password", "password", "test@test.com", "User with this email is already registered."),
 	       Arguments.of("User", "password", "passwort", "some-email@test.com", "Passwords are different!"),
 	       Arguments.of("User", "password", "password", "@test.com", "Invalid email."),
 	       Arguments.of("User", "password", "password", "some.test.com", "Invalid email."),
 	       Arguments.of("User", "pass", "pass", "some@test.com", "Password is too short.")
       );
	}
	
	@ParameterizedTest
	@MethodSource("credentials")
	public void negativeRegistrationTest(String login, String password1, String password2, String email, String error) {
		String registerError = loginPage
			.openRegistration()
			.tryRegister(login, email, password1, password2);
		
		Assert.assertEquals(error, registerError);
	}
}